docs/src/description/variables.md

```diff
@@ -34,6 +34,38 @@ integrity_constraints:
     enf a' = z[0][0] + z[0][1] + z[1][0] + z[1][1]
 ```
 
+### Syntax restriction for local variables
+Currently, it is not possible to:
+
+1. Create matrices containing both arrays and references to arrays.
+
+    Example:
+
+    ```
+    ...
+    boundary_constraints:
+        let a = [1, 2]
+        let b = [a, [3, 4]]  <-- `b` consists of array `[3, 4]` and reference to array `a`
```

Comment:
For clarity, it might be good to have examples of the things that are ok, e.g.

```
let a = [[1,2], [3,4]] # <-- this is allowed
let b = [1, 2]
let c = [3, 4]
let d = [b, c]  # <-- this is allowed
let e = [b, [3, 4]]  # <-- this is not allowed, because `b` consists of array `[3, 4]` and reference to array `a`
```

nit: I would put a `#` before the comment:

```
      let b = [a, [3, 4]]  # <-- `b` consists of array `[3, 4]` and reference to array `a`
```